fasm/parser/antlr.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env python3
```

Comment:
It is unclear to me which functions are internal methods of this module and which is the public API.

The public API should have docstrings which include things like the arguments + return types and probably a few doctest examples.

Reply:
The public API is the functions re-exported by `fasm.parser`: `parse_fasm_filename`, `parse_fasm_string`, and `implementation`.

Reply:
I have expanded the docstrings for the public functions.